### PR TITLE
feat!: decoupled builder

### PR DESCRIPTION
**Pull request by guilhem-barthes**
## Description
Merge #765 first

## Companion

* orchestrator: https://github.com/Substra/orchestrator/pull/310
* backend: https://github.com/Substra/substra-backend/pull/756
* frontend: https://github.com/Substra/substra-frontend/pull/240
* substra-generator: https://github.com/owkin/substra-generator/pull/131

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

File: backend/api/migrations/0054_alter_function_status.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+            field=models.CharField(
+                choices=[
+                    ("FUNCTION_STATUS_UNKNOWN", "Function Status Unknown"),
+                    ("FUNCTION_STATUS_WAITING", "Function Status Waiting"),
+                    ("FUNCTION_STATUS_BUILDING", "Function Status Building"),
+                    ("FUNCTION_STATUS_READY", "Function Status Ready"),
+                    ("FUNCTION_STATUS_CANCELED", "Function Status Canceled"),
+                    ("FUNCTION_STATUS_FAILED", "Function Status Failed"),
+                ],
+                max_length=64,
```
Comment (guilhem-barthes) [inline backend/api/migrations/0054_alter_function_status.py:25]: THis migration may need a data migration for the (non existing) case where someone has applied the previous migration but not this one. Alternatively, we could merge these two migrations.

File: .github/workflows/helm.yml [hunk 2]
```diff
@@ -11,6 +11,8 @@ on:
       - "main"
     paths:
       - "charts/**"
+  release:
```
Comment (ThibaultFy) [inline .github/workflows/helm.yml:14]: This file was updated to be able to publish the poc charts. Do we want to keep it ?

File: CHANGELOG.md [hunk 3]
```diff
@@ -6,13 +6,37 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
...
+- `ComputeTaskFailureReport` renamed in `AssetFailureReport` ([#727](https://github.com/Substra/substra-backend/pull/727))
+- Field `AssetFailureReport.compute_task_key` renamed to `asset_key` ([#727](https://github.com/Substra/substra-backend/pull/727))
 
 ## [0.42.2](https://github.com/Substra/substra-backend/releases/tag/0.42.2) 2023-10-18
 
 ### Changed
 
 - Update charts.
 
+### Added
```
Comment (SdgJlbl) [inline CHANGELOG.md:28]: TODO remove

File: backend/requirements.txt [hunk 4]
```diff
@@ -27,3 +27,5 @@ mozilla-django-oidc==3.0.0
 # Dependencies used in local dev mode
 argh==0.26.2
 watchdog==2.1.9
+python-dxf
+tqdm
```
Comment (SdgJlbl) [inline backend/requirements.txt:31]: Removed later in #759

File: backend/substrapp/events/reactor.py [hunk 5]
```diff
@@ -72,6 +77,40 @@ def on_computetask_event(payload):
...
+                asset_key=function_key,
+                queue=builder_queue,
+            )
+
+            building_params = {
+                "channel_name": channel_name,
+                "function_serialized": orc_function.model_dump_json(),
+            }
+            (
+                # TODO switch to function.model_dump_json() as soon as pydantic is updated to > 2.0
```
Comment (SdgJlbl) [inline backend/substrapp/events/reactor.py:105]: Remove this comment

File: backend/substrapp/migrations/0017_alter_computetaskfailurereport_logs_and_more.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+            ),
+        ),
+        migrations.AlterField(
+            model_name="functionimage",
+            name="file",
+            field=models.FileField(
+                max_length=500,
+                storage=django.core.files.storage.FileSystemStorage(),
+                upload_to=substrapp.models.function.upload_to_function,
+            ),
```
Comment (SdgJlbl) [inline backend/substrapp/migrations/0017_alter_computetaskfailurereport_logs_and_more.py:33]: @guilhem-barthes I would like to have your opinion but I'm not sure if this migration and the 0015 are not bogus migrations due to the fact that we are either using MinIO or the FileStorage depending if the deployment is local or not

File: backend/substrapp/migrations/0019_alter_assetfailurereport_logs_and_more.py [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+
+    operations = [
+        migrations.AlterField(
+            model_name="assetfailurereport",
+            name="logs",
+            field=models.FileField(
+                max_length=36,
+                storage=django.core.files.storage.FileSystemStorage(),
+                upload_to=substrapp.models.asset_failure_report._upload_to,
+            ),
```
Comment (SdgJlbl) [inline backend/substrapp/migrations/0019_alter_assetfailurereport_logs_and_more.py:24]: Same as above, potentially bogus migration

**Review by SdgJlbl — APPROVED** [review]

**Pull request merged**